src/cautiousrobot/__main__.py

```diff
@@ -207,6 +207,9 @@ def main():
         ignore = input(f"'{filename_col}' is missing values for {urls_no_name} URLs. Proceed with download ignoring these URLs? [y/n]: ")
         if ignore.lower() != "y":
             sys.exit("Exited without executing.")
+    
+    # Set source DataFrame for only non-null filename values
+    source_df = data_df.loc[data_df[filename_col].notna()].copy()
```

Comment:
I realize this is existing code moved, but does this silently skip rows with empty/null filenames? If so should we notify the user that some rows were skipped?

Reply:
I only built in a confirmation/warning to proceed if there are URLs that don't have a filename:
1. If a filename doesn't have a URL, it gets logged in the error log as response "no url".
2. If a URL doesn't have a filename, then there's an interactive choice for the user to decide if they want to proceed without downloading however many URLs don't have filenames (the number is provided) or if they want to quit and fix that first.
3. If an entry has no filename and no URL, then it is silently ignored.